src/client/systems/ClientGameController.cpp

```diff
@@ -51,7 +51,11 @@ void ClientGameController::preUpdate() {
     // call collectinputs to process window events, but if it's bot mode
     // then override any user commands
     auto inputs = m_inputController->collectInputs(m_window);
-    if (IS_BOT) {
+    // very annoying bug fixed by this % 10. When it wasn't present bot would
+    // enter a move every single time this function was called. It was buffering up
+    // to the point of being extremely dated so the bot would run off the map replaying
+    // old moves
```

Comment:
Hmm, i wonder if there is a more elegant solution

Reply:
this was a red herring. I got rid of the modulus and it runs fine.